File: PostorderTraversal.py
class TreeTraversal:
    def __init__(self, x):
        self.x = x
        self.l_side = None
        self.r_side = None


def POST_Traversal(Root_node):
   if Root_node == None:
        return
   POST_Traversal(Root_node.l_side)
   POST_Traversal(Root_node.r_side)
   print(Root_node.x)

def New(Root_node, n):
    if Root_node is None:
        Root_node = TreeTraversal(n)
        return Root_node
    if n < Root_node.x:
        Root_node.l_side = New(Root_node.l_side, n)
    else:
        Root_node.r_side = New(Root_node.r_side, n)
    return Root_node

Root_node = New(None, 32)
New(Root_node, 20)
New(Root_node, 48)
New(Root_node, 13)
New(Root_node, 21)
New(Root_node, 28)
New(Root_node, 55)
print("PostOrder-Tree-Traversal:")
POST_Traversal(Root_node)

File: PostorderTraversal.cpp
#include <bits/stdc++.h>  
using namespace std;  
  
struct node {  
    int n;  
    struct node* l_side;  
    struct node* r_side;  
};   
struct node* New(int x)  
{  
    struct node* m = (struct node*)malloc(sizeof(struct node));  
    m->n = x;  
    m->l_side = NULL;  
    m->r_side = NULL;  
  
    return (m);  
}  
void POST_Traversal(struct node* Root_node)  
{  
    if (Root_node == NULL)  
        return;  
    POST_Traversal(Root_node->l_side);  
    POST_Traversal(Root_node->r_side);  
    cout<<" "<<Root_node->n<<" ";  
}  
  
int main()  
{  
    struct node* Root_node = New(32);  
    Root_node->l_side = New(20);  
    Root_node->r_side = New(48);  
    Root_node->l_side->l_side = New(13);  
    Root_node->l_side->r_side = New(21);  
    Root_node->r_side->l_side = New(28);  
    Root_node->r_side->r_side = New(55);  
      
    cout<<"\n PostOrder-Tree-Traversal:\n";  
    POST_Traversal(Root_node);  
    return 0;  
} 
